src/TorchSharp/NN/Normalization/LayerNorm.cs

```diff
@@ -1,57 +1,72 @@
-            }
+            private long[] _normalized_shape;
+            private double _eps;
 
-            public override Tensor forward(Tensor tensor)
+            internal LayerNorm(long[] normalized_shape, double eps, bool elementwise_affine, bool bias, Device? device, ScalarType? dtype) : base(IntPtr.Zero, IntPtr.Zero)
             {
-                var res = THSNN_LayerNorm_forward(handle.DangerousGetHandle(), tensor.Handle);
-                if (res == IntPtr.Zero) { torch.CheckForErrors(); }
-                return new Tensor(res);
-            }
-
-            public Parameter? bias {
-                get {
-                    var res = THSNN_LayerNorm_bias(handle);
-                    if (res == IntPtr.Zero) { torch.CheckForErrors(); }
-                    return (res == IntPtr.Zero) ? null : new Parameter(res);
+                _normalized_shape = normalized_shape;
+                _eps = eps;
+                if (elementwise_affine)
+                {
+                    weight = Parameter(torch.empty(normalized_shape, dtype, device));
+                    if (bias)
+                    {
+                        this.bias = Parameter(torch.empty(normalized_shape, dtype, device));
+                    }
+                    else
+                    {
+                        register_parameter("bias", null);
+                    }
                 }
-                set {
-                    THSNN_LayerNorm_set_bias(handle, (value is null ? IntPtr.Zero : value.Handle));
-                    torch.CheckForErrors();
-                    ConditionallyRegisterParameter("bias", value);
+                else
+                {
+                    register_parameter("weight", null);
+                    register_parameter("bias", null);
                 }
+
+                RegisterComponents();
+
+                reset_parameters(elementwise_affine);
             }
 
-            public Parameter? weight {
-                get {
-                    var res = THSNN_LayerNorm_weight(handle);
-                    if (res == IntPtr.Zero) { torch.CheckForErrors(); }
-                    return (res == IntPtr.Zero) ? null : new Parameter(res);
+            private void reset_parameters(bool elementwise_affine)
+            {
+                if (elementwise_affine)
+                {
+                    init.ones_(weight);
                 }
-                set {
-                    THSNN_LayerNorm_set_weight(handle, value is null ? IntPtr.Zero : value.Handle);
-                    torch.CheckForErrors();
-                    ConditionallyRegisterParameter("weight", value);
+                if (bias is not null)
+                {
+                    init.zeros_(bias);
                 }
             }
+
+            public override Tensor forward(Tensor tensor)
+            {
+                return F.layer_norm(tensor, _normalized_shape, weight, bias, _eps);
+            }
+
+            public Parameter? bias { get; set; }
```

Comment:
```suggestion
            public Parameter? bias {
                get => _bias;
                set {
                    _bias?.Dispose();
                    _bias = value?.DetachFromDisposeScope() as Parameter;
                    ConditionallyRegisterParameter(nameof(bias), _bias);
                }
            }
            private Parameter? _bias;
```

Reply:
Yes -- cut-and-paste from the modified Linear.cs, and don't call RegisterComponents. 

If you want to be really efficient, then override the three `_to()` overloads and implement them directly, since you know what the parameters are (no need to call `parameters()`) That's optional, though.